### PR TITLE
itsycal: init at 0.15.3

### PR DESCRIPTION
**Pull request by 4evy**
## Description of changes

Continuation of #299257

I'm really sorry I had to open a new issue, but OfBorg gave errors for a build that works perfectly fine.

OfBorg log: https://gist.github.com/GrahamcOfBorg/bfaae351d6dc61a08283686ab23a0e80
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).
---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by stepbrobd** [comment]
The eval bug was fixed in https://github.com/NixOS/nixpkgs/pull/299288

File: pkgs/os-specific/darwin/itsycal/default.nix
Comment (stepbrobd) [inline pkgs/os-specific/darwin/itsycal/default.nix:11]: Any specific reason why `fetchurl` used but not `fetchzip`?

**Comment by DontEatOreo** [comment]
> The eval bug was fixed in #299288

Oh, nice I thought I did something wrong glad it was fix this fast as well! I guess you could ping OfBerg to eval it again, good to know for the future!

File: pkgs/os-specific/darwin/itsycal/default.nix
Comment (DontEatOreo) [reply pkgs/os-specific/darwin/itsycal/default.nix:11]: Nope, I didn't realize there was `fetchzip` will change it

Comment (DontEatOreo) [reply pkgs/os-specific/darwin/itsycal/default.nix:11]: On a second note maybe not... `fetchzip` is getting different hash and it seems like it's not downloading stuff properly either?

```console
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/ryja8i37q8kci47blxdcmjr6sm51lr9k-source
source root is .
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "configurePhase" }
configuring
no configure script, doing nothing
@nix { "action": "setPhase", "phase": "buildPhase" }
building
no Makefile or custom buildPhase, doing nothing
@nix { "action": "setPhase", "phase": "installPhase" }
installing
total 12
-rw-r--r-- 1 _nixbld1 nixbld  158 Mar 26 21:58 .sandbox.sb
-rw-r--r-- 1 _nixbld1 nixbld 5104 Mar 26 21:58 env-vars
drwxr-xr-x 3 _nixbld1 nixbld   96 Jan  1  1970 source
mv: cannot stat 'Itsycal.app': No such file or directory
```

I suggest it stays `fetchurl` unless `fetchzip` requires extra stuff I'm not aware of that would fix this? 🤔

Comment (stepbrobd) [reply pkgs/os-specific/darwin/itsycal/default.nix:11]: This worked for me (haven't tested the app tho):

```diff
diff --git a/pkgs/os-specific/darwin/itsycal/default.nix b/pkgs/os-specific/darwin/itsycal/default.nix
index 4ce7d3ae084f..8d727e703650 100644
--- a/pkgs/os-specific/darwin/itsycal/default.nix
+++ b/pkgs/os-specific/darwin/itsycal/default.nix
@@ -1,5 +1,5 @@
 { lib
-, fetchurl
+, fetchzip
 , stdenvNoCC
 , unzip
 }:
@@ -8,20 +8,16 @@ stdenvNoCC.mkDerivation (finalAttrs: {
   pname = "itsycal";
   version = "0.15.3";

-  src = fetchurl {
+  src = fetchzip {
     url = "https://itsycal.s3.amazonaws.com/Itsycal-${finalAttrs.version}.zip";
-    hash = "sha256-5aJzSuqq31B33jW4lV8vuU3eurpZBoyIW/AOC9/pxng=";
+    hash = "sha256-jpTlJY7yAARrkHzreQKbFaKj0sYp950R0qPPcDeY6AE=";
   };

-  sourceRoot = ".";
-
-  nativeBuildInputs = [ unzip ];
-
   installPhase = ''
     runHook preInstall

-    mkdir -p $out/Applications
-    mv Itsycal.app $out/Applications
+    mkdir -p $out/Applications/Itsycal.app
+    cp -R . $out/Applications/Itsycal.app

     runHook postInstall
   '';
diff --git a/pkgs/top-level/all-packages.nix b/pkgs/top-level/all-packages.nix
index c014c268962a..973238d5c8e3 100644
--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -38276,6 +38276,8 @@ with pkgs;

   itsx = callPackage ../applications/science/biology/itsx { };

+  itsycal = callPackage ../os-specific/darwin/itsycal { };
+
   iv = callPackage ../applications/science/biology/iv {
     neuron-version = neuron.version;
   };
```

Comment (stepbrobd) [reply pkgs/os-specific/darwin/itsycal/default.nix:11]: I guess the problem was due to

- `sourceRoot = ".";` (wrong location?)
- `mv Itsycal.app $out/Applications` (nix store is read-only).

Comment (DontEatOreo) [reply pkgs/os-specific/darwin/itsycal/default.nix]: Thanks for the help!

**Comment by stepbrobd** [comment]
Seems like you'll have to migrate to `pkgs/by-name`:

- remove the entry from `all-packages.nix`
- create corresponding new directories in the `pkgs/by-name` folder
- `default.nix` -> `package.nix`

**Comment by DontEatOreo** [comment]
Wait... I already have some another packages merged (#299032) without it being declared in `all-packages.nix` would I need to move that too?

Or shouldn't this package and #299032 stay in `os-specific/darwin` since that's the only platform they support?

**Comment by stepbrobd** [comment]
Sounds good, let's keep it as is.

Btw, seems like the package introduced in #299032 doesn't have an entry in `all-packages.nix`? Also, it's using `fetchurl`.

**Comment by DontEatOreo** [comment]
> Sounds good, let's keep it as is.

I'm not sure what you mean here? Do you want me to move over to `pkgs/by-name` OR stay in `os-specific/darwin` and remove the entry from `all-packages.nix`?

> Btw, seems like the package introduced in #299032 doesn't have an entry in `all-packages.nix`? Also, it's using `fetchurl`.

That's what I was saying, because if it had an entry, then it would have to be moved to `pkgs/by-name`.

I see some other packages like `raycast` for example, which is in `os-specific/darwin/` that is also in the `all-packages.nix`

File: pkgs/os-specific/darwin/itsycal/default.nix
Comment (stepbrobd) [inline pkgs/os-specific/darwin/itsycal/default.nix:29]:
```console
$ nix run nixpkgs#nixpkgs-review -- pr --print-result 299294
...
error: undefined variable 'DontEatOreo'

       at /Users/ysun/.cache/nixpkgs-review/pr-299294/nixpkgs/pkgs/os-specific/darwin/itsycal/default.nix:29:43:

           28|     license = lib.licenses.mit;
           29|     maintainers = with lib.maintainers; [ DontEatOreo ];
```

Please add an entry to https://github.com/NixOS/nixpkgs/blob/master/maintainers/maintainer-list.nix and backport it to your earliest contribution in nixpkgs where `lib.maintainers.DontEatOreo` is referenced.

I think it'd better if we involve a nixpkgs committer to see how we can resolve this, since #299032 was merged without `lib.maintainers.DontEatOreo` being defined.

Relevant: https://github.com/NixOS/nixpkgs/tree/master/maintainers

**Comment by stepbrobd** [comment]
> I'm not sure what you mean here? Do you want me to move over to `pkgs/by-name` OR stay in `os-specific/darwin` and remove the entry from `all-packages.nix`?

Sorry about my wording. Please keep `pkgs/os-specific/darwin/itsycal/default.nix` as is and leave the `itsycal` entry in `pkgs/top-level/all-packages.nix`.

The only thing we need now is to add your maintainer metadata to `maintainers/maintainer-list.nix`

File: pkgs/os-specific/darwin/itsycal/default.nix
Comment (DontEatOreo) [reply pkgs/os-specific/darwin/itsycal/default.nix:29]: I'm already there https://github.com/NixOS/nixpkgs/blob/aa1257d0d3751e2c65cc512382bdc1af1fd85c5c/maintainers/maintainer-list.nix#L5119-L5126

I guess it was a bit stupid of me, but didn't realize it was case-sensitive

Comment (stepbrobd) [reply pkgs/os-specific/darwin/itsycal/default.nix:29]: Ah, nice! I guess we would need a new PR to have the package introduced in #299032 to reference the correct maintainer (correct casing), and backport it

Comment (DontEatOreo) [reply pkgs/os-specific/darwin/itsycal/default.nix:29]: Will do tomorrow

**Comment by a-n-n-a-l-e-e** [comment]
> > I'm not sure what you mean here? Do you want me to move over to `pkgs/by-name` OR stay in `os-specific/darwin` and remove the entry from `all-packages.nix`?
> 
> Sorry about my wording. Please keep `pkgs/os-specific/darwin/itsycal/default.nix` as is and leave the `itsycal` entry in `pkgs/top-level/all-packages.nix`.
> 
> The only thing we need now is to add your maintainer metadata to `maintainers/maintainer-list.nix`

i thought that new packages all go under `by-name`

**Comment by reckenrode** [comment]
> i thought that new packages all go under `by-name`

That was my understanding too. I’d expect packages in `os-specific/darwin` to be added to `darwin-packages.nix` not `all-packages.nix`.

**Comment by stepbrobd** [comment]
> i thought that new packages all go under `by-name`

I think it'd be better to keep `itsycal` in `pkgs/os-specific/darwin/` and wait until #293498 merges, then migrate over to `by-name` with `darwin.installBinaryPackage`.

If the consensus here is to init `itsycal` in `by-name` or something else, we should follow that.

**Comment by a-n-n-a-l-e-e** [comment]
> > i thought that new packages all go under `by-name`
> 
> I think it'd be better to keep `itsycal` in `pkgs/os-specific/darwin/` and wait until #293498 merges, then migrate over to `by-name` with `darwin.installBinaryPackage`.
> 
> If the consensus here is to init `itsycal` in `by-name` or something else, we should follow that.

package guidelines are pretty clear where new apps are. let's keep this area clean. also install-binary-package is not clear it belongs here either.

**Comment by a-n-n-a-l-e-e** [comment]
another benefit is being able to take advantage of the merge-bot in `by-name` https://github.com/NixOS/nixpkgs-merge-bot

**Comment by DontEatOreo** [comment]
I guess I don't need to define `itsycal` in `all-packages.nix`?

https://github.com/NixOS/nixpkgs/actions/runs/8453647586/job/23156800994?pr=299294#step:7:31

> Such a definition is provided automatically and therefore not necessary. Please remove it.

**Comment by stepbrobd** [comment]
Result of `nixpkgs-review pr 299294` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>itsycal</li>
  </ul>
</details>

**Review by stepbrobd — APPROVED** [review]

File: pkgs/os-specific/darwin/itsycal/default.nix
Comment (drupol) [reply pkgs/os-specific/darwin/itsycal/default.nix]: Thanks for letting me know !

**Pull request merged**
